components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
+        let region_id = derived.get_id();
+
+        // Group in-memory pessimistic locks in the original region into new regions.
+        // The locks of new regions will be put into the corresponding new regions
+        // later. And the locks belonging to the old region will stay in the original
+        // map.
+        let region_locks = {
+            let mut pessimistic_locks = self.txn_ext().pessimistic_locks.write();
+            info!(self.logger, "moving {} locks to new regions", pessimistic_locks.len(); "region_id"=> region_id);
+            // Update the version so the concurrent reader will fail due to EpochNotMatch
+            // instead of PessimisticLockNotFound.
+            pessimistic_locks.version = derived_epoch.get_version();
+            pessimistic_locks.group_by_regions(&regions, derived)
+        };
+
+        // Roughly estimate the size and keys for new regions.
+        let new_region_count = regions.len() as u64;
+        let estimated_size = self.approximate_size().map(|v| v / new_region_count);
+        let estimated_keys = self.approximate_keys().map(|v| v / new_region_count);
+        let mut meta = store_ctx.store_meta.lock().unwrap();
+        meta.set_region(
+            derived.clone(),
+            self,
+            RegionChangeReason::Split,
+            tablet_index,
+        );
+        self.post_split();
+
+        // It's not correct anymore, so set it to false to schedule a split
+        // check task.
+        self.set_may_skip_split_check(false);
+
+        let is_leader = self.is_leader();
+        if is_leader {
+            self.set_approximate_size(estimated_size);
+            self.set_approximate_keys(estimated_keys);
+            // todo: this may should be doing in handle_peer_split_response.
+            self.heartbeat_pd(store_ctx);
+
+            info!(
+                self.logger,
+                "notify pd with split";
+                "region_id" => self.region_id(),
+                "peer_id" => self.peer_id(),
+                "split_count" => regions.len(),
+            );
+
+            // todo: report to PD
+        }
+
+        assert!(self.split_progress_mut().is_empty());
+        let last_key = enc_end_key(regions.last().unwrap());
+        if meta.region_ranges.remove(&last_key).is_none() {
+            panic!("{:?} original region should exist", self.logger.list());
+        }
+        let last_region_id = regions.last().unwrap().get_id();
+        for (new_region, locks) in regions.into_iter().zip(region_locks) {
+            let new_region_id = new_region.get_id();
+
+            if new_region_id == region_id {
+                let not_exist = meta
+                    .region_ranges
+                    .insert(enc_end_key(&new_region), new_region_id)
+                    .is_none();
+                assert!(not_exist, "[region {}] should not exist", new_region_id);
+                continue;
+            }
```

Comment:
`if let Some(r) = meta.regions.get(&new_region_id) {
                // Suppose a new node is added by conf change and the snapshot comes slowly.
                // Then, the region splits and the first vote message comes to the new node
                // before the old snapshot, which will create an uninitialized peer on the
                // store. After that, the old snapshot comes, followed with the last split
                // proposal. After it's applied, the uninitialized peer will be met.
                // We can remove this uninitialized peer directly.
                if util::is_region_initialized(rhttps://github.com/tikv/tikv/commit/9d87f98de78a599d13b7d10879cbee6457115127 {
                    panic!(
                        "[region {}] duplicated region {:?} for split region {:?}",
                        new_region_id, r, new_region
                    );
                }
                self.ctx.router.close(new_region_idhttps://github.com/tikv/tikv/commit/9d87f98de78a599d13b7d10879cbee6457115127;
            }`
Is this code in v1 needed in v2?